rebac-admin-backend/v1/identities_test.go

```diff
@@ -0,0 +1,585 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	qt "github.com/frankban/quicktest"
+	"go.uber.org/mock/gomock"
+
+	"github.com/canonical/identity-platform-admin-ui/rebac-admin-backend/v1/interfaces"
+	"github.com/canonical/identity-platform-admin-ui/rebac-admin-backend/v1/resources"
+)
+
+var (
+	mockFirstName  string = "MockFirstName"
+	mockIdentityId string = "test-id"
+)
+
+//go:generate mockgen -package interfaces -destination ./interfaces/mock_identities.go -source=./interfaces/identities.go
+//go:generate mockgen -package v1 -destination ./mock_error_response.go -source=./error.go
+func TestHandler_GetIdentities(t *testing.T) {
+	c := qt.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockParams := resources.GetIdentitiesParams{}
+	mockIdentityService := interfaces.NewMockIdentitiesService(ctrl)
+	mockIdentitiesReturn := resources.Identities{Data: []resources.Identity{
+		{FirstName: &mockFirstName},
+	}}
+	mockIdentityService.EXPECT().ListIdentities(gomock.Any()).Return(&mockIdentitiesReturn, nil)
+
+	expectedResponse := resources.GetIdentitiesResponse{
+		Data:   mockIdentitiesReturn.Data,
+		Status: 200,
+	}
+
+	mockWriter := httptest.NewRecorder()
+
+	sut := handler{Identities: mockIdentityService}
+	sut.GetIdentities(mockWriter, nil, mockParams)
+
+	result := mockWriter.Result()
+	defer result.Body.Close()
+
+	data, err := io.ReadAll(result.Body)
+	if err != nil {
```

Comment:
Replace this and other similar assertions with:

```go
c.Assert(err, qt.IsNil)
```

There's no need to put the "Expected error to be nil, got %v" comment, because it's obvious. But, in case you needed to add an informative comment, you can use this:

```go
c.Assert(err, qt.IsNil, qt.Commentf("blah blah: %v", err))
```